tests/test_telemetry_monitor.py

```diff
@@ -0,0 +1,1145 @@
+
+    frequency = 1
+
+    # show that an event handler w/out a monitored task will automatically stop
+    mani_handler = ManifestEventHandler("xyz", logger)
+    observer = FauxObserver()
+    duration = 2
+
+    ts0 = get_ts()
+    event_loop(observer, mani_handler, frequency, logger, cooldown_duration=duration)
+    ts1 = get_ts()
+
+    assert ts1 - ts0 >= duration
+    assert observer.stop_count == 1
+
+    # show that the new cooldown duration is respected
+    mani_handler = ManifestEventHandler("xyz", logger)
+    observer = FauxObserver()
+    duration = 5
+
+    ts0 = get_ts()
+    event_loop(observer, mani_handler, frequency, logger, cooldown_duration=duration)
+    ts1 = get_ts()
+
+    assert ts1 - ts0 >= duration
+    assert observer.stop_count == 1
+
+
+def test_telemetry_single_model(fileutils, wlmutils):
+    """Test that it is possible to create_database then colocate_db_uds/colocate_db_tcp
+    with unique db_identifiers"""
+
+    # Set experiment name
+    exp_name = "telemetry_single_model"
+
+    # Retrieve parameters from testing environment
+    test_launcher = wlmutils.get_test_launcher()
+    test_dir = fileutils.make_test_dir()
+    test_script = fileutils.get_test_conf_path("echo.py")
+
+    # Create SmartSim Experiment
+    exp = Experiment(exp_name, launcher=test_launcher, exp_path=test_dir)
+
+    # create run settings
+    app_settings = exp.create_run_settings("python", test_script)
+    app_settings.set_nodes(1)
+    app_settings.set_tasks_per_node(1)
+
+    #  # Create the SmartSim Model
+    smartsim_model = exp.create_model("perroquet", app_settings)
+    exp.generate(smartsim_model)
+    exp.start(smartsim_model, block=True)
+    assert exp.get_status(smartsim_model)[0] == STATUS_COMPLETED
+
+    telemetry_output_path = pathlib.Path(test_dir) / serialize.TELMON_SUBDIR
+    start_events = list(telemetry_output_path.rglob("start.json"))
+    stop_events = list(telemetry_output_path.rglob("stop.json"))
+
+    assert len(start_events) == 1
+    assert len(stop_events) == 1
+
+
+def test_telemetry_single_model_nonblocking(fileutils, wlmutils, monkeypatch):
+    """Ensure that the telemetry monitor logs exist when the experiment
+    is non-blocking"""
+    with monkeypatch.context() as ctx:
+        ctx.setattr(cfg.Config, "telemetry_frequency", 1)
+
+        # Set experiment name
+        exp_name = "test_telemetry_single_model_nonblocking"
+
+        # Retrieve parameters from testing environment
+        test_launcher = wlmutils.get_test_launcher()
+        test_dir = fileutils.make_test_dir()
+        test_script = fileutils.get_test_conf_path("echo.py")
+
+        # Create SmartSim Experiment
+        exp = Experiment(exp_name, launcher=test_launcher, exp_path=test_dir)
+
+        # create run settings
+        app_settings = exp.create_run_settings("python", test_script)
+        app_settings.set_nodes(1)
+        app_settings.set_tasks_per_node(1)
+
+        #  # Create the SmartSim Model
```

Comment:
Commented commented line a few times in this file. Is this intentional?

Reply:
I'm guessing not. Rm'd